src/main/java/com/projectkorra/core/util/ParticleUtil.java

```diff
@@ -0,0 +1,603 @@
+	}
+	
+	/**
+	 * Spawn an amount of colored REDSTONE particles at the given Location
+	 * @param color color of the particle
+	 * @param size size of the particle [0, ~4] (size seems to be capped around 4)
+	 * @param loc where to spawn the particle
+	 * @param amount how many of the particle to spawn
+	 */
+	public static void spawnColoredRedstone(Color color, float size, Location loc, int amount) {
+		spawnColoredRedstone(color, size, loc, amount, 0);
+	}
+	
+	/**
+	 * Spawn a single colored REDSTONE particle at the given Location
+	 * @param color color of the particle
+	 * @param size size of the particle [0, ~4] (size seems to be capped around 4)
+	 * @param loc where to spawn the particle
+	 */
+	public static void spawnColoredRedstone(Color color, float size, Location loc) {
+		spawnColoredRedstone(color, size, loc, 1);
+	}
+	
+	/**
+	 * Spawn an amount of colored REDSTONE particles at the given Location within the specified bounds and a size of 1
+	 * @param color color of the particle
+	 * @param loc where to spawn the particle
+	 * @param amount how many of the particle to spawn
+	 * @param offsets offset bounds for equal randomized spawning on each axis
+	 */
+	public static void spawnColoredRedstone(Color color, Location loc, int amount, double offsets) {
+		spawnColoredRedstone(color, 1, loc, amount, offsets, offsets, offsets);
+	}
+	
+	/**
+	 * Spawn an amount of colored REDSTONE particles at the given Location and a size of 1
+	 * @param color color of the particle
+	 * @param loc where to spawn the particle
+	 * @param amount how many of the particle to spawn
+	 */
+	public static void spawnColoredRedstone(Color color, Location loc, int amount) {
+		spawnColoredRedstone(color, 1, loc, amount, 0);
+	}
+	
+	/**
+	 * Spawn a single colored REDSTONE particle at the given Location and a size of 1
+	 * @param color color of the particle
+	 * @param loc where to spawn the particle
+	 */
+	public static void spawnColoredRedstone(Color color, Location loc) {
+		spawnColoredRedstone(color, 1, loc);
+	}
+	
+	/**
+	 * Spawn an amount of colored SPELL particles at the given Location within the specified bounds
+	 * @param color color of the particle
+	 * @param ambient if true, the particles will be considerably transparent
+	 * @param loc where to spawn the particle
+	 * @param amount how many of the particle to spawn
+	 * @param offsetX x-axis offset bounds for randomized spawning
+	 * @param offsetY y-axis offset bounds for randomized spawning
+	 * @param offsetZ z-axis offset bounds for randomized spawning
+	 */
+	public static void spawnColoredSpell(Color color, boolean ambient, Location loc, int amount, double offsetX, double offsetY, double offsetZ) {
+		if (color == null) {
+			color = Color.WHITE;
+		}
+		
+		Particle spell = ambient ? Particle.SPELL_MOB_AMBIENT : Particle.SPELL_MOB;
+		World world = loc.getWorld();
+		double r = Math.max(1, Math.min(color.getRed(), 255)) / 255D;
```

Comment:
Why is there a 1 as the first parameter of the max method here where it is 0 everywhere else?